### ISSUE TITLE
Community invite appears briefly, then shows error "failed to load"

### ISSUE THREAD

**Issue by jryans** [vector-im/riot-web#7727]
### Description

I received a community invite to +polynomial-supporters:matrix.org. When I click on it in the Riot's room list, the invite appears for ~2 secs, but it is then replaced with the error "Failed to load +polynomial-supporters:matrix.org".

In the console, I see:

```
Failed to get resource GroupMembers for +polynomial-supporters:matrix.org 
Object { errcode: "M_UNKNOWN", name: "M_UNKNOWN", message: "More than one row matched", data: {…}, httpStatus: 500, stack: "" }
GroupStore encountered error whilst fetching data for +polynomial-supporters:matrix.org 
Object { errcode: "M_UNKNOWN", name: "M_UNKNOWN", message: "More than one row matched", data: {…}, httpStatus: 500, stack: "" }
```

The failing requests seem to be:

```
GET https://matrix.org/_matrix/client/r0/groups/%2Bpolynomial-supporters%3Amatrix.org/users
[HTTP/2.0 500 Internal Server Error 942ms]
GET https://matrix.org/_matrix/client/r0/groups/%2Bpolynomial-supporters%3Amatrix.org/invited_users
[HTTP/2.0 403 Forbidden 131ms]
```

Most likely, this is related to matrix-org/synapse#2633. @benparsons suggested it could be possible for Riot ignore these errors and continue displaying the invite. I will investigate this approach.

I have sent debug logs for this issue as well.

### Steps to reproduce

- Send a community invite... possibly twice according to matrix-org/synapse#2633 (I only received the invite once that I know of, so I am unsure)
- Attempt to view the invite in Riot

I expect to be able to view and accept the invite.

### Version information

- **Platform**: web
- **Browser**: Firefox 65
- **OS**: macOS 10.13.6
- **URL**: riot.im/app

**Comment by turt2live** [comment]
This is because Riot is treating the 500 from `/users` request as a fatal error for the component (as it should - it can't reliably display the group otherwise). There was recently a patch to have the error returned from `/invited_users` work correctly, so that should not interfere with the page's ability to load.

**Comment by jryans** [comment]
> There was recently a patch to have the error returned from `/invited_users` work correctly, so that should not interfere with the page's ability to load.

Assuming you mean [the error handler in `GroupStore`](https://github.com/matrix-org/matrix-react-sdk/blob/60e22b7e0fdf7a117357b6a8813935c57cf2a090/src/stores/GroupStore.js#L145-L148), it did catch the 403 from `/invited_users`, but the 500 from `/users` is uncaught and triggers the "failed to load" message for the group. I'll experiment with also ignoring the extra error.

**Comment by jryans** [comment]
Fixed by matrix-org/matrix-react-sdk#2326.
